zeebe/gateway-rest/src/main/java/io/camunda/zeebe/gateway/rest/RestErrorMapper.java

```diff
@@ -327,4 +319,55 @@ public static ResponseEntity<Object> mapDocumentHandlingExceptionToResponse(
       final DocumentException e) {
     return mapProblemToResponse(mapDocumentHandlingExceptionToProblem(e));
   }
+
+  public static ProblemDetail mapCamundaSearchExceptionToProblem(CamundaSearchException cse) {
+    final CamundaSearchException.Reason reason =
+        (cse.getReason() == null) ? CamundaSearchException.Reason.UNKNOWN : cse.getReason();
+    final Throwable cause = (cse.getCause() == null) ? cse : cse.getCause();
+    final String title =
+        (reason == CamundaSearchException.Reason.UNKNOWN)
+            ? cause.getClass().getName()
+            : reason.name();
+    final String errorMessage = cause.getMessage();
+    final String logPrefix = "Expected to handle REST request, but error occured: ";
```

Comment:
🔧 The prefix can be a bit more generic, since it's not always an error. Plus, the placeholder can be embedded already and can be removed in all switch cases.
```suggestion
    final String logPrefix = "Expected to handle REST request, but: {}";
```